### PR TITLE
Branch dropdown menu on build dashboard

### PR DESCRIPTION
**Pull request by CaseyHillers**
This adds a dropdown button to the build dashboard appbar that allows for switching the current branch. It adds updates `FlutterBuildState` to be able to change the branch that data is shown from.

It currently erases previous branch data when switching to a new branch.

# Open Questions

I feel `moreStatusesExist` can have a better name. I am looking for a variable name that says the build state knows that more commit statuses exist from the backend, and they have to be loaded over the network. In certain cases (such as branches not master), there is only a few commits and Cocoon should not keep trying the infinite load approach

# Issues

flutter/flutter#51807 - Flutter branching for releases

# Tests

Tests for update branch and new behavior.

Widget tests to check the dropdown exists

# Preview

https://testchillers-dot-flutter-dashboard.appspot.com/#/build

### CODE REVIEW

**Comment by CaseyHillers** [comment]
/cc @Hixie if you have time, can you take a look at this? It's for adding a dropdown to the build dashboard to change branches.

I'm running into an issue where I can't get the text color to be white when the dropdown is idle, and have it be black when the dropdown menu is open.

File: app_flutter/lib/build_dashboard_page.dart [hunk 1]
```diff
@@ -119,6 +119,29 @@ class BuildDashboard extends StatelessWidget {
           title: statusTable[buildState.isTreeBuilding],
           backgroundColor: colorTable[buildState.isTreeBuilding],
           actions: <Widget>[
+            DropdownButton<String>(
+                value: buildState.currentBranch,
+                icon: Icon(
+                  Icons.arrow_downward,
+                  color: Colors.white,
```
Comment (Hixie) [inline app_flutter/lib/build_dashboard_page.dart]: in theory the color should be picked up automatically by the ambient theme, something like how i made it happen for the SIGN IN button. cc @HansMuller for advice.

File: app_flutter/lib/build_dashboard_page.dart [hunk 2]
```diff
@@ -119,6 +119,29 @@ class BuildDashboard extends StatelessWidget {
...
+                onChanged: (String branch) {
+                  buildState.updateCurrentBranch(branch);
+                },
+                items: buildState.branches.map<DropdownMenuItem<String>>((String value) {
+                  return DropdownMenuItem<String>(
+                    value: value,
+                    child: Text(value, style: TextStyle(color: Colors.black)),
+                  );
+                }).toList()),
+            Container(width: 10), // Padding between branches and sign in
```
Comment (Hixie) [inline app_flutter/lib/build_dashboard_page.dart]: `SizedBox(width: 10)` is more efficient

**Comment by Hixie** [comment]
I'm still learning the codebase, but LGTM.

File: app_flutter/lib/build_dashboard_page.dart (hunk 2 above)
Comment (CaseyHillers) [reply app_flutter/lib/build_dashboard_page.dart]: Thanks! Added.

File: app_flutter/lib/build_dashboard_page.dart (hunk 1 above)
Comment (CaseyHillers) [reply app_flutter/lib/build_dashboard_page.dart]: Thanks! I am going to submit on green to enable testing of the branching release strategy. Will follow up with smoothing this out next week.

**Pull request merged**